pkg/services/extsvcauth/extsvcaccounts/service.go

```diff
@@ -41,6 +49,46 @@ func (esa *ExtSvcAccountsService) RetrieveExtSvcAccount(ctx context.Context, org
 	}, nil
 }
 
+// SaveExternalService creates, updates or delete a service account (and its token) with the requested permissions.
+func (esa *ExtSvcAccountsService) SaveExternalService(ctx context.Context, cmd *extsvcauth.ExternalServiceRegistration) (*extsvcauth.ExternalService, error) {
+	if cmd == nil {
+		esa.logger.Warn("Received no input")
+		return nil, nil
+	}
+
+	slug := slugify.Slugify(cmd.Name)
+
+	if cmd.Impersonation.Enabled {
+		esa.logger.Warn("Impersonation setup skipped. It is not possible to impersonate with a service account token.", "service", slug)
+	}
+
+	saID, err := esa.ManageExtSvcAccount(ctx, &extsvcauth.ManageExtSvcAccountCmd{
+		ExtSvcSlug:  slug,
+		Enabled:     cmd.Self.Enabled,
+		OrgID:       extsvcauth.TmpOrgID,
+		Permissions: cmd.Self.Permissions,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// No need for a token if we don't have a service account
+	if saID <= 0 {
+		esa.logger.Debug("Skipping service account token creation", "service", slug)
+		return nil, nil
```

Comment:
isn't this an error?

Reply:
The only case you can get there is if the registration has: `Self.Enabled = false`, no permission, or no registration has been provided at all. 
In those cases, it's expected not to setup any token.